server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -184,6 +220,20 @@ public GroupShardsIterator<ShardIterator> searchShards(
         return GroupShardsIterator.sortAndCreate(new ArrayList<>(set));
     }
 
+    private void setWeightedRoundRobinAttributes(ClusterState clusterState, ClusterService clusterService) {
+        WeightedRoundRobinRoutingMetadata weightedRoundRobinRoutingMetadata = clusterState.metadata()
+            .custom(WeightedRoundRobinRoutingMetadata.TYPE);
+        this.isWeightedRoundRobinEnabled = weightedRoundRobinRoutingMetadata != null;
```

Comment:
Do we need to perform a cluster state update to persist this in cluster state else it would sit locally in-memory on the leader node.

Reply:
I am confused how the `setWeightedRoundRobinAttributes` is being invoked out of turn as a part of the search flow rather than cluster state update flow

Reply:
Weights are persisted in Cluster state with PUT call. Did you mean the same ? If weighted round robin metadata is present in cluster state then we set `isWeightedRoundRobinEnabled` to `true`

Reply:
Should we use a ClusterStateListener instead to update this dynamically?

Reply:
Updated clusterState is sent to OperationRouting during Search requests. I don't think the state will be stale. Do you have a case in mind?